graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/KotlinCodeGenTest.kt

```diff
@@ -515,6 +515,41 @@ class KotlinCodeGenTest {
         assertThat(type.propertySpecs).extracting("name").contains("genre")
     }
 
+    @Test
+    fun generateInputWithDefaultValueForEnum() {
+        val schema = """
+            enum Color {
+                red
+            }
+            
+            input ColorFilter {
+                color: Color = red
+            }
+        """.trimIndent()
+
+        val (dataTypes) = CodeGen(CodeGenConfig(schemas = setOf(schema), packageName = basePackageName, language = Language.KOTLIN)).generate() as KotlinCodeGenResult
+        assertThat(dataTypes).hasSize(1)
+
+        val data = dataTypes[0]
+        assertThat(data.packageName).isEqualTo(typesPackageName)
+
+        val members = data.members
+        assertThat(members).hasSize(1)
+
+        val type = members[0] as TypeSpec
+        assertThat(type.name).isEqualTo("ColorFilter")
+
+        val ctorSpec = type.primaryConstructor
+        assertThat(ctorSpec).isNotNull
+        assertThat(ctorSpec!!.parameters).hasSize(1)
+
+        val colorParam = ctorSpec.parameters[0]
+        assertThat(colorParam.name).isEqualTo("color")
+        assertThat(colorParam.type.toString()).isEqualTo("$typesPackageName.Color?") // FIXME: why nullable?
```

Comment:
It's nullable in the schema, the non-nullable variant would look like:

```graphql
input ColorFilter {
  color: Color! = red
}
```

Reply:
This was not intended to be committed and is removed now. Discussion is taken to #34.